arch/sim/src/sim/posix/sim_alsa.c

```diff
@@ -541,22 +551,59 @@ static int sim_audio_resume(struct audio_lowerhalf_s *dev)
 
   if (!priv->pcm)
     {
-      return 0;
+      return -ENXIO;
     }
 
   priv->paused = false;
   return 0;
 }
 #endif
 
+static int sim_audio_flush(struct audio_lowerhalf_s *dev)
+{
+  struct sim_audio_s *priv = (struct sim_audio_s *)dev;
+  int ret;
+
+  ret = nxmutex_lock(&priv->pendlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  while (!dq_empty(&priv->pendq))
+    {
+      struct ap_buffer_s *apb;
+
+      apb = (struct ap_buffer_s *)dq_remfirst(&priv->pendq);
+#ifdef CONFIG_AUDIO_MULTI_SESSION
+      priv->dev.upper(priv->dev.priv, AUDIO_CALLBACK_DEQUEUE, apb, OK, NULL);
```

Comment:
should we flush alsa driver too

Reply:
i tried snd_pcm_drop to drop pending data.
there may be no sound after snd_pcm_drop.
i think main purpose of flush is clear pendq in audio driver.  audio data which already sent to  HW may be can ignored.